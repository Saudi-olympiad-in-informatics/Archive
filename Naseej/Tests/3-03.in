167372 951
121289 11369 41529314
130543 119871 950695801
57549 105177 633034803
82734 101480 504022586
67786 47684 567324648
19724 70401 787690192
124055 69779 356412804
126474 163408 219038088
2781 111405 172709311
125323 115150 686400501
120663 125030 291924964
160978 84483 360509255
3061 152234 868943829
27783 133024 499571029
87768 30281 50792644
73695 139061 486152960
46695 99973 556179036
65798 15499 867506990
57679 93217 927667831
43229 144827 650371682
73625 77086 29937751
25261 145954 754415305
50541 114675 826135039
120577 113163 924855532
77126 105630 358686613
56964 16423 968987048
14965 161344 444994018
117101 30950 490078479
50076 72094 307788936
50372 148622 687967454
33740 24557 203740486
59146 121943 848825142
1645 86879 259232563
12933 114303 987895610
57184 137155 78246851
151089 46879 824710586
82607 92661 624549126
26561 146859 473410351
54489 40907 4032768
34800 63319 319884609
80318 134106 630810333
56454 84488 467879177
80901 21052 875047940
26124 121930 388569313
31017 26360 114106612
164692 30071 804508245
64125 39072 370978662
128718 166928 346582362
161931 81357 784114527
38847 46688 497649198
505 151615 891410920
121165 54901 691436038
108786 55482 649892159
61763 144845 803270594
99598 65530 694865277
55894 55441 938872285
4651 86095 510500390
33393 5013 323955936
144192 130827 506474949
128255 120016 593190898
59455 163886 887337478
80619 89384 346101911
1271 74434 612688741
56188 165150 728998809
61569 112194 601478121
104014 111446 279882946
133675 1767 485926432
134798 96895 828733261
132230 36337 507695712
86738 2600 726861971
123161 107458 813676823
46268 57568 555547192
87758 133659 911689309
534 48226 634787880
62515 144284 287783185
154586 148074 579440144
84667 64336 542270840
5887 92598 911968581
59485 53865 3014059
88162 128225 509658186
123392 153075 855635110
105554 42860 775820256
26639 1178 619794950
11064 27489 101914891
102493 45093 531495086
12614 87489 162126067
165266 106976 732900757
66937 126530 126812685
107848 83853 382041169
109197 65448 956840105
57794 37032 387750664
81854 152078 508362290
44127 62448 680216241
95816 160287 957625441
29700 19728 487342659
102359 70986 156236729
61012 19207 108539359
2399 56012 820307841
83912 5667 8961010
10923 152902 315636776
55053 142310 88382354
61242 33444 626553248
123198 15509 504184807
48166 92892 266782569
83455 119842 871465033
138327 105057 640763655
58662 75621 507759565
135360 154180 995121631
27787 128232 783792523
4269 138028 269256919
65711 122130 58990897
108312 65778 775366622
148826 11189 973385345
22585 68949 63266578
23487 139800 768185591
108617 116875 639666057
5019 1017 529423349
134040 27825 563143348
154557 25401 275282857
44468 47877 981133903
78858 61501 952443537
141956 3977 18959350
153473 112390 432776034
86952 66129 98656560
121636 150689 407698330
129258 137518 411010359
123738 137942 772238347
69470 35979 570022763
61857 117797 681448811
99759 3681 426511509
81955 102909 453158328
79398 142221 431711665
77104 18078 638482418
105367 70381 749942796
23986 149181 785649203
139778 113229 298605079
142202 110683 243118845
139378 117871 772228588
99287 146856 190272701
35785 128220 93446278
19174 159525 479471128
20055 40914 632544145
123704 59304 912621572
1222 17295 281197908
68730 104888 440156530
158481 161811 154198544
35955 114322 930808550
89519 85355 292224430
136115 57411 600065449
92669 40940 257166678
114278 140501 479083780
134755 157980 880978128
6790 51707 102496377
79843 39571 511349767
48294 48678 514358079
76906 107214 121410667
21846 126682 705161959
104277 154157 51891601
66640 53950 955546603
80436 52500 822703216
31951 154365 256214570
4474 135608 83626319
36390 14183 807924804
136923 118205 618325021
99437 84010 713559088
106742 90567 492286091
13203 33068 166689390
124480 153725 24627076
114633 159991 570736652
66486 27246 722159252
21929 88646 710631173
126368 50260 833654196
123465 83781 598498567
92056 26749 412790476
90637 60 274947917
61162 66808 656882145
138145 161962 188057555
110102 79763 49961299
42116 152799 21357257
89877 34669 897678508
112733 144844 668306134
118210 10021 671360576
147130 90406 452703532
69772 39996 141001018
102861 153311 205123190
151684 57869 893985666
94507 99983 769781008
92088 115738 173830473
56980 58090 735663437
152 102753 634739804
50400 76347 601838946
8113 33151 405969347
115781 146905 923881799
70667 99858 947776767
91988 32124 190771275
70830 108727 518678239
81927 96471 954127419
13953 83223 406602279
118113 18363 84421659
83313 12167 336235241
27475 83119 547737395
22108 128546 336626107
80247 128727 348695716
1516 55645 482028337
112278 53115 833122218
119390 2364 900760118
78749 156917 48226865
12189 78726 134115681
73084 99243 463324288
84771 153784 197411496
109869 150490 600413645
24183 106662 844243868
70117 111866 365906478
50675 78882 542389679
133464 104859 390372061
101448 71990 627907091
5888 147312 367105306
141233 32916 50350428
43252 18648 889510974
162583 44405 200923049
40453 39681 678773932
43688 118083 384984992
105279 110840 655261529
64572 91860 889674436
36767 37539 811146653
24099 63624 952959077
121615 14143 567128546
69983 101422 37453572
120539 80183 69780939
102311 35655 567312968
155301 9833 685037315
41338 139429 57396563
5289 116381 145612228
1244 148407 854868838
76113 20857 462780403
64424 68242 53044968
57830 23800 374595582
57247 54183 163002858
98744 61467 858265981
41123 119573 696815100
133683 97884 946275325
125374 77694 971524462
143155 25848 1833394
143451 37592 244765724
64362 40786 708247853
29051 122416 644472076
58091 101012 120370108
85382 119030 903702743
154225 20682 216366068
108982 131976 243698172
130475 14843 152725784
29872 57638 731305500
35316 73038 391105257
43658 30307 708447097
37724 44221 163304141
155436 163988 469503754
49240 91287 31932343
94917 76523 91069919
1119 2613 54373509
76905 83403 36269190
23013 61504 672764435
69636 38370 753084042
12135 44191 584561747
72622 65830 354325121
115037 156216 760659546
104237 123951 158029045
40415 130268 126789452
107056 13923 444508426
10650 81102 462854573
81805 142914 498205889
42525 42978 77436868
3625 25824 206166772
110165 82038 411148583
162806 8957 341810289
56855 6041 192926195
10052 30198 427642257
119134 114417 156227795
95817 62119 577980309
135189 15933 766798628
118039 111899 2461503
75516 145683 338900400
100377 72929 319156769
105212 145635 30237676
7334 159118 909904747
160084 40695 352520099
36186 10430 442898653
31531 77886 459710752
7699 105221 414687332
100651 35381 812024849
34394 81725 672953126
30100 53394 575751407
132044 153464 829359953
33177 137830 208447084
34303 135037 560756977
162216 136486 455223780
58127 126486 529981080
94843 164543 284014258
107713 76187 613789863
161353 20074 698247146
121760 23209 203813246
110047 77979 95611966
79337 96596 702057398
118663 70794 479695900
39416 73784 713104736
161023 91067 504332990
124065 105300 881996103
77281 13934 7573724
125091 62709 166566966
78403 62333 254536227
129076 32561 510927314
102997 110472 359827095
55819 146716 454905659
126093 73983 582775580
9918 7952 943972180
61765 77125 563973615
71957 146479 861629651
26545 155177 794236666
115655 25699 757253665
59956 133755 768319071
108878 61534 569794343
147789 5504 216836464
87782 16832 846745984
114195 40310 205173142
85992 146179 273105051
139984 78637 41790195
16970 21684 412888009
44086 113752 132781942
53196 138570 417336224
64114 99574 441322266
110788 142088 283771254
112863 10732 397766177
156897 103126 725671193
53514 30896 22421880
142004 72482 684744139
23586 35228 462034390
94849 90165 355694822
43562 41228 370635128
109751 13484 341329340
20677 71870 741578610
132381 142432 33426269
73812 56058 75696884
19981 32662 89978560
137142 112962 663818885
47836 36230 948013879
55078 56104 290164173
109764 132845 695573718
55365 161621 90222625
157860 101515 552970951
35693 92937 287938563
36149 82919 647347908
91577 115985 540856344
61625 100506 763250326
61444 11539 982170340
161149 33853 570725086
30271 104788 991143902
138954 7466 21909537
146893 7388 646200715
134681 111059 378640084
163508 36095 705754649
131298 158631 117322524
98193 32051 134559628
77387 50079 522211365
100640 54459 787335250
82498 139167 105621359
122900 108396 97017973
133638 21215 868161259
62445 138980 227555292
59745 123437 453757535
147311 23782 969395515
140495 43820 434789814
89842 124250 361916434
31814 123741 574885717
159539 145958 89640368
102416 31279 967370904
153243 109200 723695973
59117 116669 937464446
116169 106058 72419325
41240 97404 837977237
144278 119622 888450835
53778 96498 855816639
71620 93378 72040105
89484 13736 681838602
86326 70328 612593754
39860 76496 714054944
5977 20443 10630900
42164 45153 676793670
64736 80785 483949112
23859 15821 626620880
52021 35853 317682596
119679 31327 117203936
154023 129930 417019694
60930 32424 226626903
129814 38383 518192051
139618 61592 960885262
63885 90329 797956824
147091 5655 587533543
21350 133384 899088227
135626 31633 446109659
96820 72933 166796065
86138 84110 44174307
95098 9297 149307441
27957 29799 676572269
128119 127237 441013547
95728 151356 204980759
159995 54255 286817772
166366 20758 345901790
65001 78562 279073324
22377 50409 955617167
46000 69951 329605540
136628 129672 769199921
104098 115227 400913960
80709 7676 723321816
45718 116185 189338595
44142 76817 416239269
121342 82331 419696554
131215 162492 374817578
20591 126275 867977829
71059 60823 829112481
67272 43523 234805026
143521 153819 398406431
68301 88474 196208469
144692 145468 61349426
21523 63511 213713668
115503 55805 380539511
7822 68750 447024832
160011 20968 491911985
33776 85589 891904427
17132 13222 138553716
66615 120543 322909387
151194 155229 563397368
159853 137036 244792992
52063 119778 710319748
62072 63581 412860091
8405 165622 691504483
38103 9260 587540112
115142 45238 8184435
49794 93200 547081123
119528 155349 938540291
48316 16462 913167992
16264 161702 438945252
55814 74381 252893414
142345 55857 889466341
38474 84634 380953625
142724 156390 281528241
154714 74231 868975527
2224 146072 461241039
52357 112927 163262504
88481 144275 786115984
111400 7126 17308799
11968 93519 112726860
23167 110550 419092630
116240 34694 264624717
54608 62266 966680385
74327 96726 461941009
17644 71367 649635169
16980 45801 819424308
153465 144593 292226169
112937 5821 703496752
165608 29779 631116571
72645 128027 267784114
158912 37486 95846707
69563 67683 913456054
10519 156781 973980401
155874 136434 328253156
76532 90465 470244587
102760 78530 198845801
77499 61207 211083423
9093 152001 553197642
156233 78721 633185357
80795 105804 921400090
43133 69028 520447076
108788 135303 520803546
137948 12332 533775917
349 19157 577290340
52697 135680 109341178
146097 154247 961306682
33558 119187 684248942
33664 105738 741745729
137533 45316 571503227
82960 139715 221487722
160795 129652 917938816
31532 164160 395412715
103210 25283 480594155
77751 104611 894029022
123 45273 884465975
146388 121222 987000870
69452 131726 108512666
162692 31566 218651248
24698 28018 554552196
25893 41264 105138983
89903 6192 650347840
85772 34984 974580437
46062 131033 973836557
64709 9276 210112688
29349 55915 27619949
135790 20315 787026391
105575 159609 757248735
84475 150795 583688761
26649 48759 966820177
91587 125668 931929730
58254 66281 281388525
56945 54643 732400242
65627 131623 779645150
73383 7101 768942934
86773 72582 561913634
62149 99660 815590295
153026 41139 46588613
30187 37524 862105343
109407 31844 732604602
14013 131356 846597561
127065 40428 112010484
128190 38419 242236322
68187 77339 316703568
21058 126016 489988076
142985 144605 590099141
103572 144183 605279756
149409 21483 298520430
92735 71560 711896403
101239 52996 620940087
163086 128197 573116379
161249 136428 296593843
41918 78473 292939115
70791 22772 445107161
5270 159820 51701260
138734 138803 876470607
144814 46298 130923612
84541 87871 530075209
147074 127668 930061820
157344 106505 213030935
121106 162134 135402126
5398 67390 221253496
143151 110108 848528650
142365 132376 814340704
85503 81045 455989836
124852 29625 771204646
84633 94704 319225618
18926 52180 706466787
164349 140252 879312309
38338 66100 226509971
125305 106682 16447634
2516 33174 260564692
6109 10268 972211294
163974 164678 315882890
16997 25466 216278094
10383 1007 410659484
144842 139008 647143822
60220 141757 142605732
2484 70759 106197950
154458 11249 538071979
21082 75949 402066302
35596 51637 786036715
150599 43343 627043069
21174 126185 559642351
76083 69884 773279775
5668 29968 108739359
40452 79100 192672817
75893 138460 625026535
88688 133989 993432812
36106 167246 518985768
149098 132086 547402820
145530 139208 722750967
113552 95945 701880612
1599 60362 934485839
166080 77430 195171677
81929 64387 667580836
68785 159564 265191682
26957 124855 13838043
20194 34442 266766585
126296 77073 759446604
92894 16465 831193218
157183 79819 270930592
87483 109971 131123817
104773 158084 444708704
85704 7420 509459743
72784 62154 498580118
143857 164487 711795400
79094 59940 421314498
58347 101641 181444330
50482 1608 580117349
69009 145431 511307726
86545 67707 131796142
109583 29497 520965530
67789 19885 514826555
155839 96815 466490787
131665 27845 714300058
59864 152003 639328754
4790 151850 68134286
17893 147105 290099404
98002 13593 700294691
24164 150564 996704465
136061 93335 452778699
154323 114635 685179113
31791 79656 236222239
37851 59461 153655545
162696 22889 13631100
6054 133446 643512017
139154 109542 568908285
167193 126678 395580356
122542 132778 293816455
72223 80630 517265951
128391 67289 910958932
5730 74583 570900975
32509 125307 214675276
139998 55071 977726506
17170 150712 963757062
138905 92882 833696559
88337 24073 340027510
97212 43954 69616037
44160 157372 611859156
93468 533 451877538
92798 19178 629188750
125880 336 615685426
102716 19299 763828175
66761 28265 684855296
4566 32600 589412071
18072 37454 40826141
148751 142757 258876333
19719 95231 812954681
107063 152712 991932638
45471 165957 620878508
65718 82349 79702356
126027 105470 38187146
80885 91367 581330508
130635 128604 639288894
2350 140663 340484358
65719 59241 416778389
128345 42196 462786838
97303 13217 548879538
150643 91436 19770617
590 87589 441191583
163412 67809 590434847
92479 163955 283466855
111986 6771 573789857
76452 53023 278116968
157113 155184 109484739
84117 85952 230097352
133199 97424 240243943
77616 2485 799912598
23555 124358 5174061
81451 28474 360557077
105377 64644 162688372
104906 47470 380161875
13710 143513 411221775
144423 158140 582144527
74659 159664 926457011
26390 10239 895225588
134093 45773 86057814
163767 111550 610657121
105667 121966 931634897
148268 48254 804666062
40833 126864 760573016
9517 45822 789479451
134323 119193 306442104
119241 127382 84320151
119929 150415 713986074
11674 22601 19666140
144101 77948 432136736
76968 31078 670985086
8052 67409 104654563
125065 157379 813460702
153203 46754 388062235
52613 37122 311333427
29364 148671 488976507
10261 163830 824790146
34003 103819 73546859
26167 110331 632331187
97129 155465 889527279
116427 100918 755974497
13219 13893 467135989
120132 43726 262371471
18780 55758 358163013
87995 10443 310971957
21690 44968 613703845
165792 39214 85845427
38508 155548 556466400
130192 25524 639579553
56070 42247 763635744
157633 160748 656365035
62617 50712 617007437
134754 55930 918497886
84754 12686 17417026
45140 23855 448392940
16650 42172 127444970
24884 167077 392583843
3139 128056 238271862
129828 26562 502600189
147160 119458 327364049
90940 119692 783445889
58710 101327 277756103
154723 160434 938133781
25900 50417 414044274
108911 62425 131905709
122138 41361 73134688
127626 143625 468249934
69639 161484 981239546
44284 153924 998442979
15063 165869 522136414
148288 147701 192580756
47979 146847 466811617
27297 130576 469847442
127320 121435 885314327
10815 127631 470156326
141241 14665 177204859
83411 138870 799258334
147097 21693 931253685
23881 158868 467023474
122919 11421 956097704
149267 162228 346797280
114403 110541 790290290
26438 50373 262037609
127572 11866 684384817
55998 124741 479743781
100938 42078 632817955
71908 152985 583923599
76739 6000 159370021
143264 110022 613693994
112092 26824 914319009
5252 52531 528259010
34005 12615 813163370
17383 84332 255312757
143863 41527 796299621
40014 122223 496226678
77358 75686 916556456
74087 98367 507939254
11210 159310 307423784
22497 143390 408713638
6139 62928 983340638
163242 163065 976889273
132554 51380 218015785
144240 80104 298110443
41356 150011 250288605
104778 2573 733842888
133125 147993 616536573
12840 44689 130350028
647 163981 802075308
68960 113970 743348442
107296 66410 424480144
96882 147825 575711089
164666 133816 610505501
115825 126300 891888608
46276 107969 149519298
106726 88380 204090742
3879 49537 304923753
136457 107797 18352848
101254 26637 712347291
106199 52156 16163563
585 59102 624895283
34470 65996 42803592
57209 102152 790073752
6858 82163 173076557
31335 69883 566730960
103291 110559 954698376
78257 69051 478122358
104337 114061 583681008
74592 13945 21470332
47347 127515 520833112
25007 97998 55401792
69118 103755 320203651
134291 13837 799898378
75741 47986 990136441
57660 133015 334768881
34277 83897 199692171
109557 28104 571040414
73296 123194 544949567
81890 37216 766456458
83311 98517 190362322
137566 149610 762400285
128048 15583 405933487
25729 110132 370717123
84414 60546 663578246
114705 154779 388772848
94349 50358 820661635
1476 3492 675714232
75046 42368 986594643
87779 11049 468410789
93799 140964 567048126
105730 161134 810247469
129445 140875 823136597
47871 162187 675354895
144984 142400 101132220
153613 122415 93958716
137900 110942 327368349
43414 3164 568872801
8012 152482 575957421
91630 25646 172156948
22966 88739 265365242
108562 145787 491269492
150201 97617 283157168
26505 50268 907173215
71433 61539 292244490
87402 17183 656779006
72932 893 450167548
74182 81239 455632464
2321 122721 866078148
138802 45274 146261540
91396 135249 120249681
33406 72140 968622527
126338 135696 752253516
145013 160246 496199460
58478 40304 395395880
10648 84637 240279635
101163 130239 185503967
23566 31584 771450112
166190 81868 669778737
134944 37579 759307671
3842 85456 773053806
114416 86933 231822197
663 20307 806526112
36244 35204 356868345
56126 71206 743632132
39694 88570 335554091
142672 135724 388552212
124888 48054 614430317
160812 148092 160576659
113242 88980 646287207
565 94397 39374537
106271 44461 984588573
135237 95161 376752643
58704 110960 382872953
25218 35669 158741638
26152 161531 448485504
14440 122843 976705929
18448 27289 474726514
117510 75031 664004723
166816 71709 781569624
65728 140335 291372850
161052 18086 484121911
29493 140958 418587370
135262 42960 391967068
97683 143409 367651735
61359 14773 798411016
11269 17401 673877386
93846 160762 321077194
2241 84228 774879249
95134 48792 329192300
54845 163361 727788206
104764 49066 592255289
78491 111133 558114866
87835 36699 641031807
56916 75453 677204348
67174 159140 127676846
3303 51151 360062257
58664 131657 740583551
70291 95388 98276289
41479 3015 625120976
115817 24687 630052995
76457 59897 961012640
82477 24272 509919607
17146 164265 220522086
68940 164449 701257179
48576 6118 657190693
166783 53730 266996894
34151 100333 444890446
59663 89392 352914877
34634 126949 536185542
94596 22906 555297494
113354 22512 498889934
63021 61968 170078959
34597 113074 990520205
37489 155427 377859810
21832 17045 541572546
127358 30653 564988105
138682 79640 5197768
159648 28870 650519192
15738 157196 617845151
99479 57310 661345068
93625 75557 191669167
5739 85544 185022605
80258 17778 607219095
142406 101587 45223159
116397 62959 375201578
159905 22836 692742284
154398 107245 652098453
154144 35528 703743552
85602 135756 940866974
90270 141426 468340852
115352 117029 387412560
58758 14133 223306698
33407 165462 231003413
60745 44856 96079344
112229 42714 871197967
38223 16032 907723211
75461 102402 952028823
148706 83293 579587630
45368 27313 243582873
38162 45393 259506717
71079 145190 31208126
95617 129184 29303224
117393 94880 579991747
153153 151289 768731093
89136 141274 478543661
12465 30679 789623728
159577 106775 376877943
151003 57201 257488706
22826 23536 782569433
79064 35400 378999338
163431 146152 237049119
19507 36613 840576485
7984 42580 125084488
101140 57500 366417542
68713 16919 841134189
656 70030 304128676
2022 863 739455057
5808 78508 842628043
58522 133525 997988436
136155 123811 529110482
93346 32737 671768252
54302 16345 397024822
60346 138736 229391485
83829 134379 558711316
113509 98339 909203609
22440 65649 793162542
148690 114927 976170093
62083 52901 595782646
134992 54227 170604184
48775 160737 376369970
127055 44208 850762792
55622 116252 610877699
68416 121627 991812822
69592 27177 217979025
138528 11236 79140072
13114 146302 249172886
24998 114093 640100439
52725 91198 624708341
43452 63275 808153082
15108 157052 518001951
20701 66513 304167492
43792 23748 275512282
117581 99296 399703400
3262 134489 398540328
92876 119591 623585561
132175 120045 350816783
41518 58673 428946107
114232 46431 600900340
155476 13090 186765323
156662 82783 214109533
34119 117977 705570187
163154 140728 23081920
118814 143529 479645549
75547 134606 692367174
160361 6310 584151986
84752 5397 68771600
119314 128709 764259858
128500 4759 534695944
16421 117615 948129972
40349 69123 421629947
21285 123646 526606207
76899 31549 158538755
116582 41082 224932813
114055 65645 922916472
80547 157158 781145273
