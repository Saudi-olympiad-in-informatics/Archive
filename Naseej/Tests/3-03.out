78366723901953